web/js/codeworld.js

```diff
@@ -246,12 +246,23 @@ function initCodeworld() {
         docDiv.classList.add("function-tooltip-styling");
 
         const annotation = document.createElement('div');
-        renderDeclaration(annotation, functionName, keywordData, 9999, argIndex);
+        const returnedVal = renderDeclaration(annotation, functionName, keywordData, 9999, argIndex);
+        //TODO: Remove the if block once a better function parser is integrated.
+        if (returnedVal === null){
+            annotation.remove();
+            topDiv.remove();
+            return;
+        }
         annotation.className = 'hover-decl';
         docDiv.appendChild(annotation);
 
         topDiv.appendChild(docDiv);
-        window.codeworldEditor.addWidget(cursor, topDiv, true, "above", "left");
+        const lineTokens = window.codeworldEditor.getLineTokens(cursor.line);
+        const functionToken = lineTokens.filter(tkn => tkn.end <= currentToken.start && tkn.string === functionName).pop();
```

Comment:
This might be the wrong token.  For instance, if I write `rotated(rotated(solidRectangle(1,1), 45, 30)`,  the `30` token will fine the wrong `rotated` as a function name.

On the other hand, this might not find the token at all.  If I write:

    rotated(
        solidRectangle(1,1),
        45)

The function name is on a previous line.  In this case, I suspect your code with throw an exception.

Another approach might be to change the scanner again, to keep more information (e.g., line and column number of the function name) in the context.  Then you'll have it available here for use.

Reply:
Interesting, I will do that. Thanks. :-)

Reply:
I did that but found out that we didn't need the function column number and actually rendering the tooltip at the same place as the function had some issues. It made sense to have it in the same line, so didn't need the line number as well. 

Besides I think it is more important that the error/exception gets handled in the master.